### PR TITLE
fix(core): align native library naming with JNA convention

### PR DESCRIPTION
**Pull request by mtopolnik**
Currently we use `amd64` as the name for the Intel CPU architecture in the naming of our native library paths. However, JNA uses the name `x86-64`.

This is now becoming relevant because we want to integrate the output of `rust-maven-plugin` with our main build, initially for the purpose of integrating Rust into test coverage reporting. `rust-maven-plugin` respects the JNA naming convention, and it is important that is keeps doing so, because it has users outside QuestDB, who count on JNA compatibility.

In order to avoid this mismatch and/or add more complexity into the build process (copying/renaming these directories), this PR changes the naming to align with JNA.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/JitUtil.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Os.java|2|2|100.00%|

File: .github/workflows/rebuild_native_libs.yml [hunk 1]
```diff
@@ -58,15 +58,15 @@ jobs:
...
-      - name: Save darwin-amd64 Libraries to Cache
+      - name: Save darwin-x86-64 Libraries to Cache
         if: ${{ matrix.os == 'macos-13' }}
         uses: actions/cache/save@v3
         with:
           path: |
-            core/src/main/resources/io/questdb/bin/darwin-amd64/libquestdb.dylib
-            core/src/main/bin/darwin-amd64/libjemalloc.dylib
+            core/src/main/resources/io/questdb/bin/darwin-x86-64/libquestdb.dylib
+            core/src/main/bin/darwin-x8664/libjemalloc.dylib
```
Comment (ideoma) [inline .github/workflows/rebuild_native_libs.yml:67]: Why `x8664` instead of `x86-64`?

**Review by ideoma — COMMENTED** [review]
I don't see jmalloc among the renamed binaries

File: .github/workflows/rebuild_native_libs.yml (hunk 1 above)
Comment (mtopolnik) [reply .github/workflows/rebuild_native_libs.yml]: Resolved in a2f44894ddd9e8f83314f0dfa2bc72ad8cd73d4b.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/JitUtil.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Os.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/JitUtil.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Os.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/JitUtil.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Os.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/JitUtil.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Os.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/JitUtil.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Os.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/JitUtil.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Os.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/jit/JitUtil.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/Os.java|2|2|100.00%|

**Pull request merged**